compositions/find-peers.example.toml

```diff
@@ -0,0 +1,46 @@
+## THIS IS AN EXAMPLE COMPOSITION
+##
+## It performs a run of the dht/find-peers test case, with 100 instances:
+##
+##  * 10 of them are boostrappers, with upstream dependency overrides.
+##  * 45% are DHT clients.
+##  * 45% are DHT servers.
+##
+[metadata]
+name    = "find-peers-01"
+author  = "raulk"
+
+[global]
+plan    = "dht"
+case    = "find-peers"
+builder = "docker:go"
+runner  = "cluster:k8s"
+
+total_instances = 100
+
+[[groups]]
+id = "bootstrappers"
+instances = { count = 10 }
+
+  [groups.build]
+  dependencies = [
+      { module = "github.com/libp2p/go-libp2p-kad-dht", version = "995fee9e5345fdd7c151a5fe871252262db4e788"},
+      { module = "github.com/libp2p/go-libp2p", version = "76944c4fc848530530f6be36fb22b70431ca506c"},
+  ]
```

Comment:
I'm not sure if overriding dependencies is enough in the general as interfaces might have changed (also, we might want to test against javascript).

However, for our current use-case, I think it'll be fine.

---

Alternatives:

* Use a different module? That is, separate test plans from test instance runners.
* Allow build tags. That's the simplest for now.

Reply:
I dig the build tags idea; that should allow us to shim API changes elegantly. There are some edge cases where folks would need to be mindful about making a commit introducing the build tags _before_ making any changes to the actual codebase, so that the old behaviour can be shimmed and compared against the new behaviour, but that's a fair tradeoff.